### PR TITLE
Add abilitiy to configure different git hosting services.

### PR DESCRIPTION
**Pull request by cranst0n**
First run at #123. Note that I also upgraded FontAwesome to 4.7.0 only because it added a GitLab icon in 4.6.0.

### CODE REVIEW

File: src/main/scala/microsites/MicrositeKeys.scala [hunk 1]
```diff
@@ -54,5 +54,9 @@ trait MicrositeKeys {
   val micrositePalette     = settingKey[Map[String, String]]("Microsite palette")
   val micrositeGithubOwner = settingKey[String]("Microsite Github owner")
   val micrositeGithubRepo  = settingKey[String]("Microsite Github repo")
+  val micrositeGitHostingService =
+    settingKey[String]("Service used for git hosting. By default, it'll be 'Github'.")
+  val micrositeGitHostingUrl = settingKey[String](
```
Comment (raulraja) [inline src/main/scala/microsites/MicrositeKeys.scala]: can we set this automatically to empty strings when the user specifies a different `micrositeGitHostingService` automatically so they don't have to do that themselves?

**Comment by cranst0n** [comment]
@raulraja Updated with your recommendations. Also added some JS that will hide the github related watching/stars stats if it's not a GitHub project.

**Comment by cranst0n** [comment]
Is Travis is failing because it's a chicken and egg situation or is it legit? This is my first dive into an SBT plugin. It passed locally after a `publish-local`.

**Comment by raulraja** [comment]
Looks like the scripted tests are failing. They need to be adjusted to reflect the new settings.  We also need to make sure this change does not break existing configurations for other sites. Is any of the new keys or settings mandatory or change the way the setup works? Also instead of having the stats and stuff from github.com hidden perhaps we should not make the call to github.com unless that is the provider applied to a given configuration. Potentially we could add a new setting key to not fetch github.com stats and contributions which would be useful regardless for people not wanting to show those.

**Comment by cranst0n** [comment]
Right. I'm just not sure if the test I added is only failing because it's trying to fetch `0.4.1-SHAPSHOT` and coming back with the last snapshot that was published (which would not include any of the setting keys I've added) or if there's something else I'm missing/understanding. After I did a `publish-local` on my local machine, all tests passed. It looks as though my test is the only one failing on Travis.

These new settings should be optional and have no effect if they're not used.

I will update to undo hiding the GitHub stat indicators.

**Comment by cranst0n** [comment]
@raulraja Any guidance on how you want me to proceed with the test?

**Comment by raulraja** [comment]
@juanpedromoreno is there anything special that needs to happen for the scripted test to work in this scenario?

**Comment by juanpedromoreno** [comment]
@cranst0n @raulraja I've just merged this PR https://github.com/47deg/sbt-microsites/pull/126. 

Could you try again rebasing from upstream master? Thanks!

**Comment by raulraja** [comment]
@cranst0n thanks for your contribution!

**Comment by cranst0n** [comment]
@raulraja @juanpedromoreno Thanks for working to get this in. Happy to make a small contribution to a really great project!

**Pull request merged**